Dockerfile

```diff
@@ -5,33 +5,49 @@
 # Note that multistage builds can leverage the tag applied (at build
 # time) to this container
 
-FROM golang:1.12
+FROM golang:1.15 AS golauncherbuild
 LABEL maintainer="engineering@kolide.co"
 
 # fake data or not?
 ARG FAKE
 
+# Default version to build
+ARG gitver=master
+
 # The launcher build is generally not GOPATH, however, we do assume
 # that the notary files are there. Eg, we hardcode paths.
 # Look for `notaryConfigDir`
-WORKDIR /go/src/github.com/kolide/launcher
+# WORKDIR /go/src/github.com/kolide/launcher
+
+# We need the launcher source to build launcher. We can get this one
+# of three ways. git clone, copy, or mount. git clone was chosen, as
+# it's the least bad.
+# COPY is slow, and can pull in cruft from the working dir
+# mount can pull in cruft from the working dir, and can pollute the working dir
+# git clone ensures clean source
+RUN git clone https://github.com/kolide/launcher.git
+RUN cd launcher && git checkout "${gitver}"
 
 # copy source into the docker builder. Perhaps this would be cleaner
 # via a volume mount. But, COPY allows the docker container to have
 # zero impact on the host's build directory. This uses .dockerignore
 # to exclude the build directory
-COPY . ./
+# COPY . ./
```

Comment:
rm all this?

Reply:
I'd left it since I might develop it in the future, but you're right. It's trivial. cleaner to delete and re-add.